lib/web_ui/lib/src/engine/surface/path_metrics.dart

```diff
@@ -524,10 +723,275 @@ class _EllipseSegmentResult {
+  // abcd1, bcd1, cd1, p3x, p3y. Chop this part using endT;
+  final double endT = (stopT - startT) / (1 - startT);
+  final double ab2x = _interpolate(abcd1x, bcd1x, endT);
+  final double ab2y = _interpolate(abcd1y, bcd1y, endT);
+  final double bc2x = _interpolate(bcd1x, cd1x, endT);
+  final double bc2y = _interpolate(bcd1y, cd1y, endT);
+  final double cd2x = _interpolate(cd1x, p3x, endT);
+  final double cd2y = _interpolate(cd1y, p3y, endT);
+  final double abc2x = _interpolate(ab2x, bc2x, endT);
+  final double abc2y = _interpolate(ab2y, bc2y, endT);
+  final double bcd2x = _interpolate(bc2x, cd2x, endT);
+  final double bcd2y = _interpolate(bc2y, cd2y, endT);
+  final double abcd2x = _interpolate(abc2x, bcd2x, endT);
+  final double abcd2y = _interpolate(abc2y, bcd2y, endT);
+  buffer[0] = abcd1x;
+  buffer[1] = abcd1y;
+  buffer[2] = ab2x;
+  buffer[3] = ab2y;
+  buffer[4] = abc2x;
+  buffer[5] = abc2y;
+  buffer[6] = abcd2x;
+  buffer[7] = abcd2y;
+}
+
+/// Chops quadratic curve at startT and stopT and writes result to buffer.
+void _chopQuadAt(
+    List<double> points, double startT, double stopT, Float32List buffer) {
+  assert(startT != 0 || stopT != 0);
+  final double p2y = points[5];
+  final double p0x = points[0];
+  final double p0y = points[1];
+  final double p1x = points[2];
+  final double p1y = points[3];
+  final double p2x = points[4];
+
+  // If startT == 0 chop at end point and return curve.
+  final bool chopStart = startT != 0;
+  final double t = chopStart ? startT : stopT;
+
+  final double ab1x = _interpolate(p0x, p1x, t);
+  final double ab1y = _interpolate(p0y, p1y, t);
+  final double bc1x = _interpolate(p1x, p2x, t);
+  final double bc1y = _interpolate(p1y, p2y, t);
+  final double abc1x = _interpolate(ab1x, bc1x, t);
+  final double abc1y = _interpolate(ab1y, bc1y, t);
+  if (!chopStart) {
+    // Return left side of curve.
+    buffer[0] = p0x;
+    buffer[1] = p0y;
+    buffer[2] = ab1x;
+    buffer[3] = ab1y;
+    buffer[4] = abc1x;
+    buffer[5] = abc1y;
+    return;
+  }
+  if (stopT == 1) {
+    // Return right side of curve.
+    buffer[0] = abc1x;
+    buffer[1] = abc1y;
+    buffer[2] = bc1x;
+    buffer[3] = bc1y;
+    buffer[4] = p2x;
+    buffer[5] = p2y;
+    return;
+  }
+  // We chopped at startT, now the right hand side of curve is at
+  // abc1x, abc1y, bc1x, bc1y, p2x, p2y
+  final double endT = (stopT - startT) / (1 - startT);
+  final double ab2x = _interpolate(abc1x, bc1x, endT);
+  final double ab2y = _interpolate(abc1y, bc1y, endT);
+  final double bc2x = _interpolate(bc1x, p2x, endT);
+  final double bc2y = _interpolate(bc1y, p2y, endT);
+  final double abc2x = _interpolate(ab2x, bc2x, endT);
+  final double abc2y = _interpolate(ab2y, bc2y, endT);
+
+  buffer[0] = abc1x;
+  buffer[1] = abc1y;
+  buffer[2] = ab2x;
+  buffer[3] = ab2y;
+  buffer[4] = abc2x;
+  buffer[5] = abc2y;
 }
+
+double _interpolate(double startValue, double endValue, double t)
```

Comment:
nit: I'm pretty sure this already exists somewhere (`lerp`?)

Reply:
Added comment. lerpDouble uses num does null checks.